Index: ../../Libraries/open-powerplant/PowerPlant/Action Classes/LUndoer.h
===================================================================
--- ../../Libraries/open-powerplant/PowerPlant/Action Classes/LUndoer.h	(revision 15)
+++ ../../Libraries/open-powerplant/PowerPlant/Action Classes/LUndoer.h	(working copy)
@@ -46,7 +46,8 @@
 
 	virtual void		ToggleAction();
 
-	virtual void		FindUndoStatus( SCommandStatus* ioStatus );
+	virtual void		FindUndoStatus( SCommandStatus* ioStatus,
+										Boolean redo );
 
 private:					// Unimplemented copy and assignment
 						LUndoer( const LUndoer& );
