Index: ../../Libraries/MoreFilesX/MoreFilesX.c
===================================================================
--- ../../Libraries/MoreFilesX/MoreFilesX.c	(revision 1833)
+++ ../../Libraries/MoreFilesX/MoreFilesX.c	(working copy)
@@ -2332,6 +2332,7 @@
 
 /*****************************************************************************/
 
+#if 0
 OSErr
 FSVolumeMount(
 	const void *volMountInfo,
@@ -2355,6 +2356,7 @@
 
 	return ( result );
 }
+#endif
 
 /*****************************************************************************/
 
@@ -2639,6 +2641,7 @@
 
 /*****************************************************************************/
 
+#if 0
 Ptr
 GetTempBuffer(
 	ByteCount buffReqSize,
@@ -2704,6 +2707,7 @@
 
 	return ( tempPtr );
 }
+#endif
 
 /*****************************************************************************/
 
